### PR TITLE
Eliminate cross join cleanups

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestEliminateCrossJoins.java [hunk 1]
```diff
@@ -75,6 +74,21 @@ public void testEliminateSimpleCrossJoin()
                                 anyTree(ORDERS_TABLESCAN))));
     }
 
+    @Test
+    public void testDoesNotReorderJoinsWhenNoCrossJoinPresent()
+    {
+        assertPlan("SELECT * FROM (orders o1 join orders o2 on o1.orderkey = o2.orderkey) join (orders o3 join orders o4 on o3.orderkey = o4.orderkey) on o1.orderkey = o3.orderkey",
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestEliminateCrossJoins.java]: uppercase SQL keywords and add a new line before top-level `join`

File: presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestEliminateCrossJoins.java [hunk 2]
```diff
@@ -253,11 +253,6 @@ private PlanNode projectNode(PlanNode source, String symbol, Expression expressi
                 Assignments.of(new Symbol(symbol), expression));
     }
 
-    private String symbol(String name)
-    {
-        return name;
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestEliminateCrossJoins.java]: it was funny

File: presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushProjectionThroughJoin.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+/**
+ * Utility class for pushing simple projections through join so that joins can participate
+ * in cross join elimination or join reordering.
+ */
+public final class PushProjectionThroughJoin
+{
+    public static PlanNode pushProjectionThroughJoin(ProjectNode projectNode, Lookup lookup, PlanNodeIdAllocator planNodeIdAllocator)
+    {
+        if (!projectNode.getAssignments().getExpressions().stream().allMatch(DeterminismEvaluator::isDeterministic)) {
+            return projectNode;
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushProjectionThroughJoin.java]: return `Optional.empty()` (make the method `Optional`)

File: presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestEliminateCrossJoins.java [hunk 4]
```diff
@@ -36,7 +36,6 @@
     private static final PlanMatchPattern ORDERS_WITH_SHIPPRIORITY_TABLESCAN = tableScan(
             "orders",
             ImmutableMap.of("O_ORDERKEY", "orderkey", "O_SHIPPRIORITY", "shippriority"));
-    private static final PlanMatchPattern SUPPLIER_TABLESCAN = tableScan("supplier", ImmutableMap.of("S_SUPPKEY", "suppkey"));
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestEliminateCrossJoins.java]: separate commit?

File: presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushProjectionThroughJoin.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            else {
+                checkState(rightChild.getOutputSymbols().contains(requiredSymbol));
+                rightAssignmentsBuilder.putIdentity(requiredSymbol);
+            }
+        }
+
+        Assignments leftAssignments = leftAssignmentsBuilder.build();
+        Assignments rightAssignments = rightAssignmentsBuilder.build();
+        List<Symbol> outputSymbols = Streams.concat(leftAssignments.getOutputs().stream(), rightAssignments.getOutputs().stream())
+                .filter(symbol -> projectNode.getOutputSymbols().contains(symbol))
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushProjectionThroughJoin.java]: `projectNode.getOutputSymbols()::contains` 
or perhaps `ImmutableSet.copyOf(projectNode.getOutputSymbols())::contains`

**Comment by sopel39** [comment]
This needs more work

**Pull request closed without merging**